x-pack/plugins/ingest_manager/public/applications/ingest_manager/index.tsx

```diff
@@ -39,86 +41,175 @@ export const ProtectedRoute: React.FunctionComponent<ProtectedRouteProps> = ({
+                      defaultMessage="You are not authorized to access Ingest Manager. Ingest Manager requires {roleName} privileges."
+                      values={{ roleName: <EuiCode>superuser</EuiCode> }}
+                    />
+                  ) : (
+                    <FormattedMessage
+                      id="xpack.ingestManager.securityRequiredErrorMessage"
+                      defaultMessage="You must enable security in Kibana and Elasticsearch to use Ingest Manager."
+                    />
+                  )}
+                </p>
+              }
+            />
+          </Panel>
+        )}
+      </ErrorLayout>
+    );
+  }
+
   if (!isInitialized || initializationError) {
     return (
-      <EuiErrorBoundary>
-        <DefaultLayout>
-          <WithoutHeaderLayout>
-            {initializationError ? (
-              <Error
-                title={
-                  <FormattedMessage
-                    id="xpack.ingestManager.initializationErrorMessageTitle"
-                    defaultMessage="Unable to initialize Ingest Manager"
-                  />
-                }
-                error={initializationError}
+      <ErrorLayout>
+        {initializationError ? (
+          <Error
+            title={
+              <FormattedMessage
+                id="xpack.ingestManager.initializationErrorMessageTitle"
+                defaultMessage="Unable to initialize Ingest Manager"
               />
-            ) : (
-              <Loading />
-            )}
-          </WithoutHeaderLayout>
-        </DefaultLayout>
-      </EuiErrorBoundary>
+            }
+            error={initializationError}
+          />
+        ) : (
+          <Loading />
+        )}
+      </ErrorLayout>
     );
   }
 
   return (
-    <EuiErrorBoundary>
-      <Router {...rest}>
-        <Switch>
-          <ProtectedRoute path={EPM_PATH} isAllowed={epm.enabled}>
-            <DefaultLayout section="epm">
-              <EPMApp />
-            </DefaultLayout>
-          </ProtectedRoute>
-          <Route path={AGENT_CONFIG_PATH}>
-            <DefaultLayout section="agent_config">
-              <AgentConfigApp />
-            </DefaultLayout>
-          </Route>
-          <Route path={DATA_STREAM_PATH}>
-            <DefaultLayout section="data_stream">
-              <DataStreamApp />
-            </DefaultLayout>
-          </Route>
-          <ProtectedRoute path={FLEET_PATH} isAllowed={fleet.enabled}>
-            <DefaultLayout section="fleet">
-              <FleetApp />
-            </DefaultLayout>
-          </ProtectedRoute>
-          <Route exact path="/">
-            <DefaultLayout section="overview">
-              <IngestManagerOverview />
-            </DefaultLayout>
-          </Route>
-          <Redirect to="/" />
-        </Switch>
-      </Router>
-    </EuiErrorBoundary>
+    <PackageInstallProvider notifications={notifications}>
+      <FleetStatusProvider>
```

Comment:
These two providers were moved here from `IngestManagerApp` so that they are loaded only after permissions & setup checks are done